test/parallel/test-tls-writewrap-leak.js

```diff
@@ -12,13 +12,15 @@ const server = net.createServer(common.mustCall((c) => {
   c.destroy();
 })).listen(0, common.mustCall(() => {
   const c = tls.connect({ port: server.address().port });
-  c.on('error', () => {
-    // Otherwise `.write()` callback won't be invoked.
-    c._undestroy();
-  });
+
+  c.on('error', common.mustCall((err) => {
+    assert.strictEqual(err.code, 'ECONNRESET');
+    server.close();
+  }));
 
   c.write('hello', common.mustCall((err) => {
-    assert.strictEqual(err.code, 'ECANCELED');
+    // TODO
+    // assert.strictEqual(err.code, 'ECANCELED');
```

Comment:
I'm not sure how to interpret this assertion. We do allow writes to succeed after destroy.